client/service/src/task_manager/mod.rs

```diff
@@ -124,10 +155,11 @@ impl SpawnTaskHandle {
 			}
 		};
 
-		let join_handle = {
-			let _span = self.telemetry_span.as_ref().map(|s| s.enter());
-			self.executor.spawn(Box::pin(future.in_current_span()), task_type)
-		};
+		let future = future.in_current_span().boxed();
+		let join_handle = self.executor.spawn(
```

Comment:
Did you explore instrumenting the executor itself? Not sure if that would work, but if it did maybe it could perhaps save some busywork?

Reply:
In the end that is what we are doing here. We instrument the executor :D 

The `TaskExecutor` itself is just a `Fn(Future, Type)`